test/tests.spec.ts

```diff
@@ -715,17 +715,118 @@ describe("node-saml /", function () {
+            TEST_CERT +
+            '</ds:X509Certificate></ds:X509Data></ds:KeyInfo></ds:Signature><saml:Subject><saml:NameID Format="urn:oasis:names:tc:SAML:2.0:nameid-format:transient">ploer@subspacesw.com</saml:NameID><saml:SubjectConfirmation Method="urn:oasis:names:tc:SAML:2.0:cm:bearer"><saml:SubjectConfirmationData NotOnOrAfter="2014-05-28T00:19:08Z" Recipient="{recipient}" InResponseTo="_a6fc46be84e1e3cf3c50"/></saml:SubjectConfirmation></saml:Subject><saml:Conditions NotBefore="2014-05-28T00:13:08Z" NotOnOrAfter="2014-05-28T00:19:08Z"><saml:AudienceRestriction><saml:Audience>{audience}</saml:Audience></saml:AudienceRestriction></saml:Conditions><saml:AuthnStatement AuthnInstant="2014-05-28T00:16:07Z" SessionNotOnOrAfter="2014-05-29T00:16:08Z" SessionIndex="_30a4af50-c82b-0131-f8b5-782bcb56fcaa"><saml:AuthnContext><saml:AuthnContextClassRef>urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport</saml:AuthnContextClassRef></saml:AuthnContext></saml:AuthnStatement></saml:Assertion>' +
+            "</samlp:Response>";
+          const base64xml = Buffer.from(xml).toString("base64");
+          const container = { SAMLResponse: base64xml };
+          try {
+            const samlObj = new SAML({ ...noAudienceSamlConfig, audience: "no match" });
+            const { profile } = await samlObj.validatePostResponseAsync(container);
+            expect(profile).to.not.exist;
+          } catch (err: unknown) {
+            expect(err).to.exist;
+            expect(err).to.be.instanceOf(Error);
+            expect((err as Error).message).to.match(/SAML assertion audience mismatch/);
+          }
+        });
+
+        it("if audience not provided, fail", async () => {
```

Comment:
This is a bit of "twister". Documentation says that if `audience` is not provided value of audience defaults to value of options' `issuer`. More precise testcase description would be "if issuer and audience not provided, fail" (which is not exact definition either because issuer would default to "onelogin_saml" and "audience" would default to "unknown_audience" (see earlier comment).

IMHO This testcase would not be needed if ties to onelogin would be cut off and `issuer` would be changed to required argument (and audience would default to `issuer` if it is not defined).

Reply:
I've updated the name of the test. I agree that the test won't be needed after the change to make `issuer` mandatory. I'll make sure to remove this test as part of that change.